### PR TITLE
feat(background-media): add shadow parts to bg-media

### PR DESCRIPTION
**Pull request by bruno-amorim**
### Related Ticket(s)

[ADCMS-5036](https://jsw.ibm.com/browse/ADCMS-5036)

### Description

Adding shadow parts to Background Media component so that it can be styled outside the shadow tree.

### Changelog

Shadow parts added to Background Media. 
Snapshots updated

### CODE REVIEW

**Review by jkaeser — CHANGES_REQUESTED** [review]
Good call updating the unit test snapshots!

The only critique I have is that your part names don't have a consistent style... some are using the block + element part of BEM naming (e.g. `video-player__controls`), and some are using block + modifier (e.g. `background-media--item`). Can you make these consistent?

File: packages/web-components/src/components/background-media/background-media.ts [hunk 1]
```diff
@@ -145,6 +145,7 @@ class C4DBackgroundMedia extends C4DImage {
 
     return html`
       <button
+        part="video-player__controls"
```
Comment (andy-blum) [inline packages/web-components/src/components/background-media/background-media.ts:148]:
```suggestion
        part="controls"
```

File: packages/web-components/src/components/background-media/background-media.ts [hunk 2]
```diff
@@ -156,7 +157,11 @@ class C4DBackgroundMedia extends C4DImage {
   }
 
   renderGradient() {
-    return html` <div class="${this._getGradientClass()}"></div> `;
+    return html`
+      <div
+        part="background-media--gradient"
```
Comment (andy-blum) [inline packages/web-components/src/components/background-media/background-media.ts:162]:
```suggestion
        part="gradient"
```

File: packages/web-components/src/components/background-media/background-media.ts [hunk 3]
```diff
@@ -186,9 +191,12 @@ class C4DBackgroundMedia extends C4DImage {
 
   render() {
     return html`
-      <div class="${this._getMobilePositionClass()}">
+      <div
+        part="background-media--container"
```
Comment (andy-blum) [inline packages/web-components/src/components/background-media/background-media.ts:195]:
```suggestion
        part="container"
```

File: packages/web-components/src/components/background-media/background-media.ts [hunk 4]
```diff
@@ -186,9 +191,12 @@ class C4DBackgroundMedia extends C4DImage {
...
+        class="${this._getMobilePositionClass()}">
         ${this.gradientHidden ? '' : this.renderGradient()}
         <div
+          part="background-media--item"
```
Comment (andy-blum) [inline packages/web-components/src/components/background-media/background-media.ts:199]:
```suggestion
          part="item"
```

**Review by andy-blum — CHANGES_REQUESTED** [review]

**Comment by andy-blum** [comment]
See https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/11796#pullrequestreview-2066279339

**Review by jkaeser — APPROVED** [review]

**Review by andy-blum — APPROVED** [review]

**Pull request merged**